SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -207,6 +213,41 @@ void ItemPrinterRNG::value_changed(void* object){
         AFTER_ITEM_PRINTER_DONE_EXPLANATION.set_text("Then stop the program.");
         MATERIAL_FARMER_DISABLED_EXPLANATION.set_text("<br>To enable the Material Farmer, enable \"Automatic Material Farming\" above");
     }
+
+    if (overlapping_bonus()){
+        OVERLAPPING_BONUS_WARNING.set_visibility(ConfigOptionState::ENABLED);
+        // std::cout << "Warning: One of the Ball/Item bonuses haven't been fully used up. This can mess up your rewards for subsequent prints. Note: Each Ball/Item bonus lasts for 10 prints" << std::endl;
+    }else{
+        OVERLAPPING_BONUS_WARNING.set_visibility(ConfigOptionState::HIDDEN);
+    }
+    
+}
+
+
+// - return true if the user is trying to trigger another Ball/Item bonus without
+// using up the previous bonus. Doing this can mess up rewards for subsequent prints, 
+// since a ball bonus interferes with activating an item bonus and vice versa.
+// - each Bonus lasts for 10 prints.
+bool ItemPrinterRNG::overlapping_bonus(){
+    // for each row in table. if ball/item bonus, ensure that sum of prints in subsequent rows >=10 before the next bonus, or end of the table.
+    uint16_t total_prints_since_last_bonus = 10;
+    for (std::shared_ptr<EditableTableRow> table_row : TABLE0.current_refs()){
+        std::shared_ptr<ItemPrinterRngRow> row = std::static_pointer_cast<ItemPrinterRngRow>(table_row);

```

Comment:
It's actually safe to take the reference directly here since the lifetime is already preserved in the outer loop.

`ItemPrinterRngRow& row = static_cast<ItemPrinterRngRow&>(*table_row);`

Reply:
I can't get your suggestion to compile.
From my reading, shared_ptr can only be cast with static_pointer_cast.